cmd/soroban-cli/src/strval.rs

```diff
@@ -120,21 +122,21 @@ pub fn from_json(v: &Value, t: &ScSpecTypeDef) -> Result<ScVal, StrValError> {
         }
         (ScSpecTypeDef::I32, Value::Number(n)) => ScVal::I32(
             n.as_i64()
-                .ok_or(StrValError::InvalidValue)?
+                .ok_or(StrValError::InvalidValue(Some(ScSpecType::I32)))?
```

Comment:
💡 I think if we pass in `t`, which is an `ScSpecTypeDef`, instead of writing out an `ScSpecType` enum value, we'll get even more information in the cases that types have sub types, generics, etc.

Reply:
I agree, but using `ScSpecTypeDef` added complexity for `Vec`, `Map`, `Tuple`, etc and also custom types and I wanted a first-approach working solution.

Reply:
I had a look and switched it to using def. It doesn't result in additional functional, but it should make ti easier to maintain because we already have the def in all cases so we don't need to hardcode types and instead can pass the def value through.